bwe/RateController.cpp

```diff
@@ -611,7 +656,24 @@ size_t RateController::getPacingBudget(uint64_t timestamp) const
             static_cast<uint32_t>(
                 _model.bandwidthKbps * (timestamp - _model.lastTransmission) / (8 * utils::Time::ms)));
 
-    return _model.targetQueue - std::min(_model.targetQueue, currentQueue);
+    auto currentTargetQ = _model.targetQueue;
+    if (_probe.isProbing(timestamp))
+    {
+        currentTargetQ += _config.mtu;
+        currentTargetQ += static_cast<uint32_t>(_model.targetQueue * (timestamp - _probe.start) / _config.probeRampup);
```

Comment:
We said when we probe we want to double the targetQ. But we can go much further than that.
A probe can last 700ms. if _config.probeRampup is 300ms, the we have the double queue at 300ms . And we will have 3x more at 600ms. Shouldn't we limit it to the double? or isn't a problem having such high budget?

Reply:
True we are not limiting and that is perhaps why the bw probes worked better now.
The risk is that we exceed the buffer capacity of the network and induce loss. The is not a risk when we are under estimating since the drain rate is much higer than we think. But it is a problem when we are already near capacity. Then the probe will cause build up and then loss, and it will take much longer time to reduce the delay. It can only drain as fast as we are underutilizing the link. If we have 98% utilization the delay will remain for a very long time.